### PR TITLE
Feature/webcaster device selection

### PR DESCRIPTION
**Pull request by fh-nano**

### CODE REVIEW

File: docs/webrtc/nanostream_webrtc_device_selection.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+Also we will create a Javascript function that populates above device selection lists later on.
+
+```js
+// adds options to a select element
+var createSelectOptions = function (elementId, devices) {
+
+  var selectOptions = document.getElementById(elementId).options;
+
+  // Add all devices by id and index
+  for (var device of devices) {
```
Comment (nanocosmos-aa) [inline docs/webrtc/nanostream_webrtc_device_selection.md:34]: > var device of devices

Shouldn't it be `in`?

File: docs/webrtc/nanostream_webrtc_device_selection.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+// request device list, will fire 'ReceivedDeviceList' event
+user.getDevices();
+```
+
+## Start the preview
+
+Once a user has selected audio and video devices from the lists, the preview can be started.
+
+```js
+// we will trigger the breview once the 'start-preview-button' has been clicked.
```
Comment (nanocosmos-aa) [inline docs/webrtc/nanostream_webrtc_device_selection.md:69]: > breview

Typo!

File: docs/webrtc/nanostream_webrtc_device_selection.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+document.getElementById('start-preview-button').addEventListener('click', function() {
+
+  // get the index of the selected audio device
+  var audioDeviceList = document.getElementById("audio-device-list");
```
Comment (nanocosmos-aa) [inline docs/webrtc/nanostream_webrtc_device_selection.md:73]: > "audio-device-list"

We use single quotes :)

File: docs/webrtc/nanostream_webrtc_device_selection.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  var audioDeviceIndex = audioDeviceList.options[audioDeviceList.selectedIndex].value;
+
+  // get the index of the selected video device
+  var videoDeviceList = document.getElementById("video-device-list");
```
Comment (nanocosmos-aa) [inline docs/webrtc/nanostream_webrtc_device_selection.md:77]: > "video-device-list"

Same here

File: docs/webrtc/nanostream_webrtc_device_selection.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+sidebar_label: Device Selection
+---
+
+The Webcaster API gives you the possibility to present all currently attached audio and video devices to your users.
+The API call to requrest the device list is [getDevices()](../nanostream_webrtc_api/#getdevices).<br>
+You will receive the list of devices in the [ReceivedDeviceList](../nanostream_webrtc_api/#RtcUser+event_ReceivedDeviceList) event. After that you can show this list to the user, so he can choose which devices he wants to be used for the Webcast.
+
+## Preparation
+
+We will need two `<select>` tags in our HTML document, in order to display the attached audio & video devices in your HTML document. Also we will add a button to trigger the [startPreview(previewConfig)](../nanostream_webrtc_api/#startpreviewconfig) call after we made our device selection in th UI and a video element to show the preview in.
```
Comment (nanocosmos-aa) [inline docs/webrtc/nanostream_webrtc_device_selection.md:13]: > th UI

Typo!

File: docs/webrtc/nanostream_webrtc_device_selection.md (hunk 1 above)
Comment (fh-nano) [reply docs/webrtc/nanostream_webrtc_device_selection.md:34]: no, "for...of" gives you the entries of iterables, devices is an array

File: docs/webrtc/nanostream_webrtc_device_selection.md (hunk 2 above)
Comment (fh-nano) [reply docs/webrtc/nanostream_webrtc_device_selection.md]: done

File: docs/webrtc/nanostream_webrtc_device_selection.md (hunk 3 above)
Comment (fh-nano) [reply docs/webrtc/nanostream_webrtc_device_selection.md]: done

File: docs/webrtc/nanostream_webrtc_device_selection.md (hunk 4 above)
Comment (fh-nano) [reply docs/webrtc/nanostream_webrtc_device_selection.md]: done

File: docs/webrtc/nanostream_webrtc_device_selection.md (hunk 5 above)
Comment (fh-nano) [reply docs/webrtc/nanostream_webrtc_device_selection.md]: done

**Pull request merged**